src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -278,7 +279,7 @@ private Activity StartActivity(HttpContext httpContext, out bool hasDiagnosticLi
             if (_diagnosticListener.IsEnabled(ActivityStartKey))
             {
                 hasDiagnosticListener = true;
-                _diagnosticListener.StartActivity(activity, new { HttpContext = httpContext });
+                StartActivity(activity, httpContext);
```

Comment:
I like this and we may be able to update listener in time for 3.0 though. But it is breaking (i.e. there is no HttpContext property on HttpContext type), everyone who attempts to get it will fail.

Reply:
ooops, you've added property on HttpConetxt :)

Reply:
HttpContext.HttpContext 😄